VstsClientLibrariesSamples/GettingStarted/Authentication.cs

```diff
@@ -32,8 +82,51 @@ public IEnumerable<TeamProjectReference> PersonalAccessToken(string url, string
                 else
                 {
                     return null;
-                }                
+                }
             }
         }
+
+        internal static string GetAccountTenant(string vstsAccountName)
+        {
+            string tenant = null;
+            using (var client = new HttpClient())
+            {
+                client.BaseAddress = new Uri(String.Format("https://{0}.visualstudio.com", vstsAccountName));
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+                client.DefaultRequestHeaders.Add("User-Agent", "VSTSAuthSample-AuthenticateADALNonInteractive");
+                client.DefaultRequestHeaders.Add("X-TFS-FedAuthRedirect", "Suppress");
+                HttpResponseMessage response = client.GetAsync("_apis/connectiondata").Result;
+
+                try { 
+                    // Get the tenant from the Login URL
+                    var wwwAuthenticateHeaderResults = response.Headers.WwwAuthenticate.ToList();
+                    var bearerResult = wwwAuthenticateHeaderResults.Where(p => p.Scheme == "Bearer");
+                    foreach (var item in wwwAuthenticateHeaderResults)
+                    {
+                        if (item.Scheme.StartsWith("Bearer"))
+                        {
+                            tenant = item.Parameter.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries)[2];
+                            break;
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // In MSA backed accounts, there is no tenant
+                    return null;
+                }
+            }
+
+            if (String.IsNullOrEmpty(tenant))
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Something went wrong...");
```

Comment:
Please provide additional information about the failure retrieving the tenant, like the wwwAuthenticateHeaderResults or something. Otherwise, the error message isn't actionable.